src/docker/dockerclient.cc

```diff
@@ -10,7 +10,24 @@ DockerClient::HttpClientFactory DockerClient::DefaultHttpClientFactory = []() {
   return std::make_shared<HttpClient>("/var/run/docker.sock");
 };
 
-DockerClient::DockerClient(bool curl, const HttpClientFactory& http_client_factory) {
+DockerClient::DockerClient(std::shared_ptr<HttpInterface> http_client) : http_client_{std::move(http_client)} {}
+
+bool DockerClient::serviceRunning(const std::string& app, const std::string& service, const std::string& hash) {
+  updateContainerStatus();
```

Comment:
ah - you've address my previous comment. You should probably squash these two commits .

Reply:
yep, I think I will update the previous commit with this change and remove it from the commit related to testing so both commits are self-sufficient and don't mix up topics.

Reply:
In addition, to this change, in the future, I am going to add another method to AppEngine interface (e.g. refreshStat()) so the docker daemon is inquired once per an update cycle not per each App as it is now.